### PR TITLE
008 Script for requirement [PerformAudioPassThru] requested "ttsChunks" is NOT supported

### PR DESCRIPTION
**Pull request by maria-mih**
008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua 
All common functions used available at #703
Testing coverage available at #643
@istoimenova Please review

### CODE REVIEW

**Comment by istoimenova** [comment]
@maria-mih, please reference script in the TestSet.

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+-- valid <audioPassThruIcon> parameter
+-- and the requested <audioPassThruIcon> exists at app`s sandbox (see AppStorageFolder section)
+-- as well as another related to request valid mandatory params
+-- SDL must transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--1. Used preconditions
+--1.1 All parameters are present and within bounds speech capabilities is sent with type = "SILENCE "
+--1.2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, other params, ttsChunksType = "SILENCE") from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params, ttsChunksType = "SILENCE") to HMI
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih, please clarify why you test ttsChunksType = "SILENCE"?

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+--2.2 SDL sends TTS.Speak to HMI
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih, ttsName(ttsChunksType) is parameter of TTS.Speak

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih, points 2.3-2.4 are redundant in description.

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+--2.4 HMI sends TTS.Speak (UNSUPPORTED_RESOURCE) to SDL
+-- Expected result:
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih , expected result should be:
SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
SDL sends TTS.Speak to HMI

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+-- SDL sends PerformAudioPassThru (WARNINGS, success:true) to mobile app
+
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih, please remove empty row.

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih, please remove space before brackets. Apply this for all calls of functions / require.

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+commonSteps:PutFile("Precondition_PutFile_With_Icon","icon.png")
+
+function Test:Precondition_Check_audioPassThruIcon_Existence()
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih, please move parameters on previous row.

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_SILENCE()
+  testCasesForPerformAudioPassThru:PerformAudioPassThru_Diff_Speech_Capabilities (self, "SILENCE")
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih, as the idea of atomic script is to show what exactly is happening, will be better to declare the TestStep in script, not to call common function.
Calling of common functions is useful as preconditions/postconditions.

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_SILENCE()
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @maria-mih , please be sure that RPC PerformAudioPassThru is allowed by policy.

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 5 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 6 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 7 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 9 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

File: test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua (hunk 8 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/008_ATF_P_PerformAudioPassThru_WARNINGS_Different_Speech_Capabilities_SILENCE.lua]: @istoimenova Please check in commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

**Comment by maria-mih** [comment]
@istoimenova Script has been updated after first review iteration. Please check on commit 58e095bb7a883682ed6d10c29cf71406a691a1ec

**Pull request merged**